source/dagger/gameplay/team_game/golem.cpp

```diff
@@ -39,18 +38,11 @@ void ancient_defenders::GolemBehaviorSystem::Run()
 			else if (golem_.currentAction == EAction::Moving) {
 				auto nextPosition = golem_.postition - 1;
 
-				auto coords = WalkingPath::path.back();
-				int i = 0;
-				for (auto iter : WalkingPath::path) {
-					if (i == nextPosition) {
-						coords = iter;
-						break;
-					}
-					i++;
-				}
+				auto coords = WalkingPath::path[nextPosition];
+				
 
-				auto destinationX = coords.x; // Maybe add +/- a few percent to make paths bit more varied
-				auto destinationY = coords.y;
+				auto destinationX = coords.x + golem_.offset.x;
```

Comment:
might be good to rename golem_ to enemy_ too

Reply:
Forgot to change it here as well, it will be changed.